include/SDL_video_capture.h

```diff
@@ -0,0 +1,301 @@
+ * The value can be used to fill SDL_VideoCaptureSpec structure.
+ *
+ * \param id video capture opened device id
+ * \param index format between 0 and num -1
+ * \param format pointer output format (SDL_PixelFormatEnum)
+ * \param type pointer output type
+ *
+ * \returns 0 on success, negative value on error
+ *
+ * \sa SDL_GetNumVideoCaptureFormats
+ */
+extern DECLSPEC int SDLCALL SDL_GetVideoCaptureFormat(SDL_VideoCaptureDeviceID id, int index, Uint32 *format, Uint32 *type);
+
+/**
+ * Number of available formats for the device
+ *
+ * \param id video capture opened device id
+ *
+ * \returns number of formats, negative value on error
+ *
+ * \sa SDL_GetVideoCaptureFormat
+ */
+extern DECLSPEC int SDLCALL SDL_GetNumVideoCaptureFormats(SDL_VideoCaptureDeviceID id);
+
+/**
+ * Get frame sizes of the device and the specified input format.
+ * The value can be used to fill SDL_VideoCaptureSpec structure.
+ *
+ * \param id video capture opened device id
+ * \param format a format that can be used by the device (SDL_PixelFormatEnum)
+ * \param index framesize between 0 and num -1
+ * \param width output width
+ * \param height output height
+ *
+ * \returns 0 on success, negative value on error
+ *
+ * \sa SDL_GetNumVideoCaptureFrameSizes
+ */
+extern DECLSPEC int SDLCALL SDL_GetVideoCaptureFrameSize(SDL_VideoCaptureDeviceID id, Uint32 format, int index, Uint32 *width, Uint32 *height);
+
+/**
+ * Number of different framesizes available for the device and pixel format.
+ *
+ * \param id video capture opened device id
+ * \param format frame pixel format (SDL_PixelFormatEnum)
+ *
+ * \returns number of framesize, negative value on error
+ *
+ * \sa SDL_GetVideoCaptureFrameSize
+ */
+extern DECLSPEC int SDLCALL SDL_GetNumVideoCaptureFrameSizes(SDL_VideoCaptureDeviceID id, Uint32 format);
+
+
+/**
+ * Get video capture status
+ *
+ * \param id video capture opened device id
+ *
+ * \returns 0 on success, negative value on error
+ *
+ * \sa SDL_VideoCaptureStatus
+ */
+extern DECLSPEC SDL_VideoCaptureStatus SDLCALL SDL_GetVideoCaptureStatus(SDL_VideoCaptureDeviceID id);
+
+/**
+ * Start video capture
+ *
+ * \param id video capture opened device id
+ *
+ * \returns 0 on success, negative value on error
+ *
+ * \sa SDL_StopVideoCapture
+ */
+extern DECLSPEC int SDLCALL SDL_StartVideoCapture(SDL_VideoCaptureDeviceID id);
+
+/**
+ * Acquire a frame.
+ * The frame is a memory pointer to the image data, whose size and format
+ * are given by the the obtained spec.
+ *
+ * After used, the frame should be released with SDL_VideoCaptureReleaseFrame
+ *
+ * \param id video capture opened device id
+ * \param frame pointer to get the frame
+ * \param ticks pointer to get the ticks when this was read from driver
+ *
+ * \returns 0 on success, negative value on error
+ *
+ * \sa SDL_VideoCaptureReleaseFrame
+ */
+extern DECLSPEC int SDLCALL SDL_VideoCaptureAcquireFrame(SDL_VideoCaptureDeviceID id, SDL_VideoCaptureFrame *frame, Uint64 *ticks);
```

Comment:
I've just been looking at the header, I haven't looked at implementation. One important distinction between different video capture APIs is blocking vs non blocking. I assume if you call this function and the webcam hasn't "generated" a new image, it will block until it give you a new frame?

Reply:
no, this isn't not blocking. that may change. 
but making a blocking api is on top is easier.
I've updated the header

Reply:
That's cool, I think a non blocking API seems more user friendly. 

So it might return the same frame pixel data on multiple calls?

I don't see any header updates on this, I'm thinking it should explicitly say it doesn't block if there isn't a new frame available, it does `X`.

pygame.camera has a `query_image()` function that returns True if a new frame is ready. That seems very helpful for a non blocking api, you can only UpdateTexture or convert to a surface when needed.